File: typescript/eagleEye.ts
import * as fs from 'fs';
import * as path from 'path';
import * as screenshot from 'screenshot-desktop';
import { createWorker } from 'tesseract.js';

const BASE_DIR = __dirname;
const SCREENSHOT_PATH = path.join(BASE_DIR, 'screenshot.png');
const OUTPUT_PATH = path.join(BASE_DIR, 'output.txt');
const LOG_PATH = path.join(BASE_DIR, 'error.log');

const logError = (message: string) => {
    const timestamp = new Date().toISOString();
    fs.appendFileSync(LOG_PATH, `${timestamp}: ${message}\n`);
};

console.log("Preparing to take screenshot in 5 seconds...");
setTimeout(async () => {
    try {
        const imgPath = await screenshot({ filename: SCREENSHOT_PATH });
        console.log("Screenshot captured successfully.");

        const worker = createWorker();
        await worker.load();
        await worker.loadLanguage('eng');
        await worker.initialize('eng');

        const { data: { text } } = await worker.recognize(imgPath);
        if (text) {
            fs.writeFileSync(OUTPUT_PATH, text);
        } else {
            logError("Failed to extract text with Tesseract.");
        }

        await worker.terminate();
    } catch (error) {
        logError(`Failed to capture screenshot and extract text: ${error}`);
    }
}, 5000);

File: python/eagleEye.py
import os
import time
import subprocess
from PIL import ImageGrab
from pytesseract import image_to_string

BASE_DIR = os.path.dirname(os.path.realpath(__file__))
SCREENSHOT_PATH = os.path.join(BASE_DIR, 'screenshot.png')
OUTPUT_PATH = os.path.join(BASE_DIR, 'output.txt')
LOG_PATH = os.path.join(BASE_DIR, 'error.log')

def log_error(message):
    with open(LOG_PATH, 'a') as f:
        f.write(f"{time.strftime('%Y-%m-%d %H:%M:%S')}: {message}\n")

print("Preparing to take screenshot in 5 seconds...")
time.sleep(5)

img = ImageGrab.grab()
img.save(SCREENSHOT_PATH)

print("Screenshot captured successfully.")

text = image_to_string(img)

if text:
    with open(OUTPUT_PATH, 'w') as f:
        f.write(text)
else:
    log_error("Failed to extract text with Tesseract.")